src/test/java/com/arpnetworking/metrics/incubator/impl/SafeRefLockTest.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright 2017 Inscope Metrics, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.incubator.impl;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests for the {@link SafeRefLock} class.
+ *
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+ */
+public class SafeRefLockTest {
+
+    @Test
+    public void testMultipleReadersOk() throws InterruptedException {
+        final Semaphore waitForBoth = new Semaphore(0);
+        final Semaphore releaseBoth = new Semaphore(0);
+        final Object testObject = new Object();
+        final SafeRefLock<Object> refLock = new SafeRefLock<>(testObject);
+
+        final Runnable r = () -> {
+            refLock.readLocked((object) -> {
+                waitForBoth.release();
+                try {
+                    releaseBoth.acquire();
+                } catch (final InterruptedException ignored) {
+                }
+            });
+        };
+
+        final Thread thread1 = new Thread(r);
```

Comment:
In a test? Why not? I'm explicitly testing concurrency and I need to know how and when these things get executed.